paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/ManageScreenInteractor.kt

```diff
@@ -78,7 +92,20 @@ internal class DefaultManageScreenInteractor(
 
     private fun handlePaymentMethodSelected(paymentMethod: DisplayableSavedPaymentMethod) {
         onSelectPaymentMethod(paymentMethod)
-        navigateBack()
+        safeNavigateBack()
+    }
+
+    private fun noPaymentMethodsAvailableToManage(displayableSavedPaymentMethods: List<DisplayableSavedPaymentMethod>): Boolean {
+        val onlyOneNotModifiablePm = displayableSavedPaymentMethods.size == 1 &&
+            !allowsRemovalOfLastSavedPaymentMethod &&
+            !displayableSavedPaymentMethods.first().isModifiable()
+        return displayableSavedPaymentMethods.isEmpty() || onlyOneNotModifiablePm
+    }
+
+    private fun safeNavigateBack() {
+        if (!hasNavigatedBack.getAndSet(true)) {
+            navigateBack()
+        }
```

Comment:
This is necessary because we are calling this from within `combineAsStateFlow` which gets called multiple times. If we go back multiple times, we'll end up closing paymentSheet entirely